From 385386ef390c0f67c3b86625c4acdde64b27945f Mon Sep 17 00:00:00 2001
Message-Id: <385386ef390c0f67c3b86625c4acdde64b27945f.1431284209.git.chang-joon.lee@intel.com>
In-Reply-To: <89460f3fc3d6c0659d44897a6adec3c3fecc5564.1431284209.git.chang-joon.lee@intel.com>
References: <89460f3fc3d6c0659d44897a6adec3c3fecc5564.1431284209.git.chang-joon.lee@intel.com>
From: Deepak S <deepak.s@intel.com>
Date: Tue, 24 Mar 2015 13:43:08 +0530
Subject: [PATCH 32/68] Revert "SQUASHME! [VPG]: drm/i915: Remove pin & unpin
 fence for gen8+"

Right fix will be pulled from upstream.

This patch reverts commit:
commit 9804fdf64d75e149a63cba9bcb0e403f683e6260
Change-Id: I2aea7772586daacd5a08ef157621c638aea27a57
Author: Deepak S <deepak.s@intel.com>
Date:   Tue Mar 3 19:07:09 2015 +0530

    SQUASHME! [VPG]: drm/i915: Remove pin & unpin fence for gen8+

For: GMINL-7369
Change-Id: Iaecf6f7b3d5883cbcdce00db629844715c8af2ea
Signed-off-by: Deepak S <deepak.s@intel.com>
---
 drivers/gpu/drm/i915/intel_display.c |    7 ++-----
 1 file changed, 2 insertions(+), 5 deletions(-)

diff --git a/drivers/gpu/drm/i915/intel_display.c b/drivers/gpu/drm/i915/intel_display.c
index 904d06a..62665b1 100644
--- a/drivers/gpu/drm/i915/intel_display.c
+++ b/drivers/gpu/drm/i915/intel_display.c
@@ -2377,10 +2377,9 @@ intel_pin_and_fence_fb_obj(struct drm_device *dev,
 		ret = i915_gem_object_get_fence(obj);
 		if (ret)
 			goto err_unpin;
-
-		i915_gem_object_pin_fence(obj);
 	}
 
+	i915_gem_object_pin_fence(obj);
 	drm_gem_object_reference(&obj->base);
 
 	dev_priv->mm.interruptible = true;
@@ -2395,9 +2394,7 @@ err_interruptible:
 
 void intel_unpin_fb_obj(struct drm_i915_gem_object *obj)
 {
-	struct drm_i915_private *dev_priv = obj->base.dev->dev_private;
-	if ((INTEL_INFO(dev_priv->dev)->gen <= 7))
-		i915_gem_object_unpin_fence(obj);
+	i915_gem_object_unpin_fence(obj);
 	i915_gem_object_unpin_from_display_plane(obj);
 	drm_gem_object_unreference(&obj->base);
 }
-- 
1.7.9.5

